src/main/java/sparksoniq/spark/iterator/flowr/LetClauseSparkIterator.java

```diff
@@ -36,25 +38,113 @@
 import java.util.List;
 
 public class LetClauseSparkIterator extends FlowrClauseSparkIterator {
+
+    private VariableReferenceIterator _variableReference;
+    private String _variableName;           // for efficient use in local iteration
+    private RuntimeIterator _expression;
+    private DynamicContext _tupleContext;   // re-use same DynamicContext object for efficiency
+    private FlworTuple _nextLocalTupleResult;
+
     public LetClauseSparkIterator(RuntimeTupleIterator child, VariableReferenceIterator variableReference, RuntimeIterator expression, IteratorMetadata iteratorMetadata) {
         super(child, null, FLWOR_CLAUSES.LET, iteratorMetadata);
         this._children.add(variableReference);
         this._children.add(expression);
+        _variableReference = variableReference;
+        _expression = expression;
     }
-    
+
+    @Override
+    public boolean isRDD() {
+        if (this._child == null) {
+            return false;
+        } else {
+            return _child.isRDD();
+        }
+    }
+
+    @Override
+    public FlworTuple next() {
+        if(_hasNext == true){
+            FlworTuple result = _nextLocalTupleResult;      // save the result to be returned
+            setNextLocalTupleResult();              // calculate and store the next result
+            return result;
+        }
+        throw new IteratorFlowException("Invalid next() call in let flwor clause", getMetadata());
+    }
+
+    private void setNextLocalTupleResult() {
+        // if first let clause, there are no more tuples
+        if (this._child == null) {
+            this._hasNext = false;
+            return;
+        }
+
+        if (_child.hasNext()) {
+            FlworTuple inputTuple = _child.next();
+            _tupleContext.removeAllVariables();             // clear the previous variables
+            _tupleContext.setBindingsFromTuple(inputTuple);      // assign new variables from new tuple
+
+            List<Item> result = new ArrayList<>();
```

Comment:
For performance, this should be moved to a field in the class. That way, the object is only created once in the LetClauseSparkIterator constructor, and, at this position of the code, you just need to empty it. This saves all the object creations, which are very expensive.